src/app/modules/services/pull-action-protocol/pull-action-protocol.service.spec.ts

```diff
@@ -0,0 +1,207 @@
+import { HttpClient, HttpClientModule, HttpErrorResponse } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { fakeAsync, inject, TestBed, tick } from '@angular/core/testing';
+import { MessagingModule, MessagingService } from '@testeditor/messaging-service';
+import { Conflict } from './conflict';
+import { PullActionProtocol } from './pull-action-protocol.service';
+import { HttpProviderService } from '../http-provider-service/http-provider.service';
+
+export const HTTP_STATUS_CONFLICT = 409;
+let actionCalledCount = 0;
+function  getActionsFor(...answers: any[]): (client: HttpClient) => Promise<Boolean> {
+  actionCalledCount = 0;
+  return (client: HttpClient) => {
```

Comment:
… took me a while to figure out what this function is actually doing…! I don't have any concrete improvement suggestions, but maybe add a short comment on its intention? I was struggling with understanding one of the tests (the one that repeats three times), because I was wondering **why** the third pass suddenly succeeded – until I realized that the third _action_ was actually a different one.